pkg/services/serviceaccounts/proxy/service.go

```diff
@@ -19,70 +19,77 @@ import (
 type ServiceAccountsProxy struct {
 	log            log.Logger
 	proxiedService serviceaccounts.Service
+	isProxyEnabled bool
 }
 
 func ProvideServiceAccountsProxy(
+	features *featuremgmt.FeatureManager,
 	proxiedService *manager.ServiceAccountsService,
 ) (*ServiceAccountsProxy, error) {
 	s := &ServiceAccountsProxy{
 		log:            log.New("serviceaccounts.proxy"),
 		proxiedService: proxiedService,
+		isProxyEnabled: features.IsEnabled(featuremgmt.FlagExternalServiceAccounts),
 	}
 	return s, nil
 }
 
 var _ serviceaccounts.Service = (*ServiceAccountsProxy)(nil)
 
 func (s *ServiceAccountsProxy) AddServiceAccountToken(ctx context.Context, serviceAccountID int64, cmd *serviceaccounts.AddServiceAccountTokenCommand) (*apikey.APIKey, error) {
-	sa, err := s.proxiedService.RetrieveServiceAccount(ctx, cmd.OrgId, serviceAccountID)
-	if err != nil {
-		return nil, err
-	}
-
-	if isExternalServiceAccount(sa.Login) {
-		s.log.Error("unable to create tokens for external service accounts", "serviceAccountID", serviceAccountID)
-		return nil, extsvcaccounts.ErrCannotCreateToken
+	if s.isProxyEnabled {
+		sa, err := s.proxiedService.RetrieveServiceAccount(ctx, cmd.OrgId, serviceAccountID)
+		if err != nil {
+			return nil, err
+		}
+
+		if isExternalServiceAccount(sa.Login) {
+			s.log.Error("unable to create tokens for external service accounts", "serviceAccountID", serviceAccountID)
+			return nil, extsvcaccounts.ErrCannotCreateToken
+		}
 	}
 
 	return s.proxiedService.AddServiceAccountToken(ctx, serviceAccountID, cmd)
 }
 
 func (s *ServiceAccountsProxy) CreateServiceAccount(ctx context.Context, orgID int64, saForm *serviceaccounts.CreateServiceAccountForm) (*serviceaccounts.ServiceAccountDTO, error) {
-	if !isNameValid(saForm.Name) {
-		s.log.Error("Unable to create service account with a protected name", "name", saForm.Name)
-		return nil, extsvcaccounts.ErrInvalidName
+	if s.isProxyEnabled {
+		if !isNameValid(saForm.Name) {
```

Comment:
Nit.
```suggestion
	if s.isProxyEnabled && !isNameValid(saForm.Name) {
```

Reply:
I thought about this but decided against, this is code that will be removed after all.